langchain/vectorstores/azuresearch.py

```diff
@@ -171,21 +181,74 @@ def __init__(
         search_type: str = "hybrid",
         semantic_configuration_name: Optional[str] = None,
         semantic_query_language: str = "en-us",
+        fields: list[SearchField] = None,
+        vector_search: VectorSearch = None,
+        semantic_settings: SemanticSettings = None,
+        scoring_profiles: Optional[list[ScoringProfile]] = None,
+        default_scoring_profile: Optional[str] = None,
         **kwargs: Any,
     ):
+        from azure.search.documents.indexes.models import (
+            PrioritizedFields,
+            SearchableField,
+            SearchField,
+            SearchFieldDataType,
+            SearchIndex,
+            SemanticConfiguration,
+            SemanticField,
+            SemanticSettings,
+            SimpleField,
+            VectorSearch,
+            VectorSearchAlgorithmConfiguration,
+        )
         """Initialize with necessary components."""
         # Initialize base class
         self.embedding_function = embedding_function
+        default_fields = [
+            SimpleField(
+                name=FIELDS_ID,
+                type=SearchFieldDataType.String,
+                key=True,
+                filterable=True,
+            ),
+            SearchableField(
+                name=FIELDS_CONTENT,
+                type=SearchFieldDataType.String,
+                searchable=True,
```

Comment:
For the `SearchableField` class you don't need to set `searchable` to `True` since it's already set by default.

Same for the other SearchableField references below